utils/versions.py

```diff
@@ -27,15 +27,12 @@ def get_version(collection_id: Union[int, str], uri: str) -> str:
     Returns a version path.
     """
     collection_id = str(collection_id)
-    uri = uri.rstrip('/')
-    uri_parts = uri.lstrip('/').split('/')
+    uri_parts = uri.strip('/').split('/')
     if str(collection_id) not in uri_parts or len(uri_parts) < 2:
         raise Exception(f"Not a valid version path: {uri}, {uri_parts}")
-    rikolti_data_root, relative_path = uri.split(f"{collection_id}/")
-    path_list = relative_path.split('/')
+    path_list = uri_parts[uri_parts.index(collection_id):]
```

Comment:
```suggestion
    path_list = uri_parts[uri_parts.index(f"{collection_id}/"):]
```

Okay, so this screwed me over at one point: I had to include the `/` because sometimes collection 233 has a `233/vernacular_v1/data/233.json` page number. 

How you have it implemented here should be fine, because `index` should snag the _first_ instance of `collection_id`, but the suggestion makes it a bit more explicit.

Reply:
Oh, actually, I think this isn't quite right, either. `data` needs to be a discrete path part, not just in the path somewhere. `233/vernacular_v1/mapped_metadata_v1/` includes `data`.

Reply:
So we actually do want to split on `/` and then check if `data` is in the path list.

Reply:
Was there something broken with this implementation of `get_version`? It was pretty tricky to implement, so if it's not broken, I don't think it's worth fixing.

Reply:
Hrm, the way I have it implemented, `f"{collection_id}/` isn't going to be in `path_list` because we're splitting on `/`.

`path_list` is going to look something like:
```
["8", "vernacular_metadata_2024-01-31T00:39:58", "data", "986"]
```

So we just need to find the index of `{collection_id}`